Fixed typos, spelling correction

dartraiden opened the pull request.

dartraiden commented:
In \src\core\dnsupdater.cpp (line 177) remains unclear comment " // Pase reply". Probably meant "Pass" or "Pause"

pmzqla commented:
> // Pase reply

I guess it's "Parse reply"

sledgehammer999 commented:
Thank you for the effort. If you actually went through and read it manually-> WOW. (I suspect you ran the file through a spell checker instead).
Some files are 3rd party files and I don't want to make changes there.
Plus old entries in the changelog shouldn't be changed now. Too late.

So I want you to exclude these:
1.  Changelog
2. m4/ax_boost_base.m4
3. m4/pkg.m4
4. src/app/qtsingleapplication/qtlockedfile.cpp
5. src/searchengine/nova/fix_encoding.py

dartraiden commented:
Yes, I read sources manially :) I reverted unwanted changes and and fixed typo from https://github.com/qbittorrent/qBittorrent/pull/2938#issuecomment-98555220. Thanks @pmzqla for the tip

sledgehammer999 commented:
Squash the commits into one. Hint use `git rebase -i <hash>` and then choose "fixup"

dartraiden commented:
Done

sledgehammer999 commented:
Apparently you didn't read #3134.
Can you rebase your PR on current master and have clean history?
If you want this to be applied on the v3_2_x make a separate PR targeting that branch and having also a clean history.
If you don't know how to use git, tell me where you're stuck so I can provide you with instructions.
Thanks for your effort in fixing typos.

sledgehammer999 commented:
Also "squash" your commits into one. (I can give instructions on how to do that with git)

dartraiden commented:
Sorry, I'm not very good at working with git.
I'm trying to "git rebase -i 6e4fbcf017e6b77473a231d4ea272c35cad85478" pick my first commit 9cbab5e73d5f4072a79668087bd142552a0bcb7f and fixup others, but it gives an error "couldnt apply f1bce0b... Fix downloadthread* coding style"

What should I do to rebase my PR on current master?

dartraiden commented:
In the extreme case I can make a new PR and remove the PR. It will be easy, since all typos already known, I can find them by searching in the file manager

sledgehammer999 commented:
For future reference: When you make develop on a branch and it changes upstream. Don't do "git merge <upstreambranch>" do merge the upstream changes. Do instead "git rebase <upstreambranch>. This will make git replay YOUR commits AFTER the commits of the upstream branch.

The problem here is with the merge commit you did and messed up the history.
First create a copy of this branch: `git -b fix_typos master`.
Now let's have your master be an exact copy of OUR master. First you need to discard all your changes.
Switch to master `git checkout master`
Discard your changes: `git reset --hard 6e4fbcf017e6b7747`
Now bring your master up to speed with out master `git pull --ff-only upstream`. Change `upstream` to whatever name you have given to our repo.
Now let's cherry-pick your individual commits and hope the can be applied automatically on top of the new history.
1. `git cherry-pick 9cbab5e73d5f4072a79668087bd142552a0bcb7f`
2. `git cherry-pick 8e3b1edc53333858efc251195684340561689a48`
3. `git cherry-pick 1169099ddf64c05b957123099c673d90ab72630f`

Some of the above cherry-picks may fail. Try invoking the mergetool and it may auto-solve them `git mergetool`. If you don't have a mergetool configured git will complain. If you have it may autoresolve the conflict or you will be presented with several options. I think you'll understand what to do in this case.
After you're done with the mergetool(and every conflict has been resolved) issue `git mergetool --continue`
After that continues to the next cherry-pick command.

After all cherry-picks are done, you can easily do `git rebase -i <hash>` and squash them into one.

Then you just do `git push --force origin master` to update your branch and this PR.

Don't delete the `fix_typos` branch. You can use the same steps to make a PR for the v3_2_x branch.

If you're stuck with the cherry-picks(and don't know how to resolve the conflicts), then I don't think I'll be able to help you more. Unfortunately the only realistic solution will be to redo the changes...

dartraiden commented:
git -mergetool worked great, but seems that It seems that I made a mistake with git rebase...

sledgehammer999 commented:
If am not mistaken you only have one commit now. So let's fix the things you borked with rebase -i.

Create a copy of your master "git checkout -b fix_typos2 master`
Go to master again: `git checkout master`
Drop wrong commits: `git reset --hard e01a61d816e58e4`
Bring your master up to speed with out master `git pull --ff-only upstream master` (where `upstream` the name you have for our repo`
Just cherry-pick your commit on top: `git cherry-pick bd7d3dfb7ac066511ee9b5ba57a26c508ddd9cd9`
(this cherry-pick shouldn't have conflicts)
Update this PR: `git push --force origin master`

dartraiden commented:
It seems, now everything is in place. Thank you very much!

sledgehammer999 commented:
@glassez `json_parser.hh` and `json_scanner.cc` were generated automatically, right?

sledgehammer999 commented:
@dartraiden `excanvas-compressed.js` seems to be a 3rd party file, so don't fix typos there. (wiat first to see if you should exclude json_parser.* too)

glassez commented:
>@glassez json_parser.hh and json_scanner.cc were generated automatically, right?

These files are part of QJson library and shouldn't be changed.
And yes. These files were generated automatically.

sledgehammer999 commented:
@dartraiden exclude `excanvas-compressed.js`, `json_parser.hh` and `json_scanner.cc` and then I can merge it.

dartraiden commented:
Done

sledgehammer999 commented:
Thank you very much.
